.github/workflows/build.yml

```diff
@@ -0,0 +1,35 @@
+name: build
+on:
+  push:
+    branches:
+      - master
+      - /\d\.0\.0-RC/
+  pull_request:
+    branches:
+      - master
+      - /\d\.0\.0-RC/
+jobs:
+  check-coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Set branch name and PR number
+        id: refs
+        env:
+          BRANCH_NAME_OR_REF: ${{ github.head_ref || github.ref }}
+        run: |
+          echo "::set-output name=branch_name::${BRANCH_NAME_OR_REF#refs/heads/}"
+          echo "::set-output name=pr_number::$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")"
+      - name: Coveralls Maven plugin
```

Comment:
This does not only run "Coveralls Maven plugin". It runs complete build with all the checks, compilation and tests + generates javadocs + generates and sends coverage report using that mentioned plugin